pandas/core/apply.py

```diff
@@ -133,8 +134,14 @@ def get_result(self):
         elif isinstance(self.f, np.ufunc):
             with np.errstate(all='ignore'):
                 results = self.f(self.values)
-            return self.obj._constructor(data=results, index=self.index,
-                                         columns=self.columns, copy=False)
+            result = self.obj._constructor(data=results, index=self.index,
```

Comment:
I think we'd like to avoid going to sparse in the first place. This seems to be taking the result and converting it back to sparse if necessary.

Reply:
I haven't looked closely at where things are going wrong though, so I'm not sure where you should be looking.

Reply:
The bug report seems to specifically target the case where the user chooses to store sparse columns in a normal `DataFrame` instead of a `SparseDataFrame`, for instance by calling the `DataFrame` constructor with a `SparseDataFrame` as the data.

Reply:
Understood. It's a deeper issue though.

A ufunc on a SparseArray will stay sparse. It won't blow up your memory.

```python
In [15]: df = pd.DataFrame({"A": pd.SparseArray([0, 1, 2])})

In [16]: np.exp(df.A.values)
Out[16]:
[1.0, 2.718281828459045, 7.38905609893065]
Fill: 1.0
IntIndex
Indices: array([1, 2], dtype=int32)
```

But a ufunc (possibly via apply) on a DataFrame or Series with sparse values will materialize the dense  values:

```python
In [18]: np.exp(df.A).dtypes
Out[18]: dtype('float64')
```

Just converting back to sparse so that the output dtype is `Sparse[float64]` isn't enough, your memory has already blown up :)

I think we need to ensure that ufuncs on Series dispatch to the underlying array.

Reply:
Is this better? `self.obj[col].values` returns a `SparseArray` so the ufunc should stay sparse.